src/replication/replication_manager.cpp

```diff
@@ -0,0 +1,273 @@
+#include "replication/replication_manager.h"
+
+#include <chrono>  // NOLINT
+#include <fstream>
+#include <optional>
+
+#include "common/error/exception.h"
+#include "common/json.h"
+#include "loggers/replication_logger.h"
+#include "network/network_io_utils.h"
+#include "storage/recovery/replication_log_provider.h"
+#include "storage/write_ahead_log/log_io.h"
+
+namespace {
+
+bool CompareJsonBuffer(nlohmann::json left, nlohmann::json right) { return left.at("buf_id") > right.at("buf_id"); }
+
+}  // namespace
+
+namespace noisepage::replication {
+
+Replica::Replica(common::ManagedPointer<messenger::Messenger> messenger, const std::string &replica_name,
+                 const std::string &hostname, uint16_t port)
+    : replica_info_(messenger::ConnectionDestination::MakeTCP(replica_name, hostname, port)),
+      connection_(messenger->MakeConnection(replica_info_)),
+      last_heartbeat_(0) {}
+
+ReplicationManager::ReplicationManager(
+    common::ManagedPointer<messenger::Messenger> messenger, const std::string &network_identity, uint16_t port,
+    const std::string &replication_hosts_path,
+    common::ManagedPointer<common::ConcurrentBlockingQueue<storage::BufferedLogWriter *>> empty_buffer_queue)
+    : messenger_(messenger),
+      identity_(network_identity),
+      port_(port),
+      provider_(std::make_unique<storage::ReplicationLogProvider>(std::chrono::seconds(1))),
+      received_buffer_queue_(CompareJsonBuffer),
+      empty_buffer_queue_(empty_buffer_queue) {
+  // TODO(WAN): provider_ arguments
```

Comment:
oh, and I need to fix this to not be a constant. But that's somewhat lumped together with the replication log provider.

Reply:
Is this referring to the 1 second timeout for the provider? You could just add it as a setting or toss it in `include/common/constants.h